# importing dependencies

In [147]:
import numpy as np
import pandas as pd
import datetime
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.cluster import KMeans
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler

# loading the data 

In [148]:
customers= pnd.read_csv('C:/Users/ASUS/Desktop/datasets/customer.csv',encoding='ISO_8859-1' ,delimiter=';')
products= pnd.read_csv('C:/Users/ASUS/Desktop/datasets/products.csv',encoding='ISO_8859-1' ,delimiter=';')
sales= pnd.read_csv('C:/Users/ASUS/Desktop/datasets/sales.csv',encoding='ISO_8859-1' ,delimiter=';')

# Cleaning Customers dataset

In [149]:
customers.head()

id  country  gender  age       profession
0  CT-0001   France  Female   42       graduation
1  CT-0002  Tunisia  Female   41            basic
2  CT-0003  Tunisia  Female   42            basic
3  CT-0004    Egypt  Female   39       graduation
4  CT-0005  Algeria  Female   43  undergraduation

In [150]:
customers.describe()

age
count  1000.00000
mean     39.12600
std      10.29933
min      18.00000
25%      32.00000
50%      37.00000
75%      44.00000
max      82.00000

In [151]:
customers.shape

(1000, 5)

#### ====> The table of customers have 1000 rows and 6 columns.

### informations on features

In [152]:
customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          1000 non-null   object
 1   country     1000 non-null   object
 2   gender      1000 non-null   object
 3   age         1000 non-null   int64 
 4   profession  1000 non-null   object
dtypes: int64(1), object(4)
memory usage: 39.2+ KB


### Missing values

In [153]:
customers = customers.dropna()
print("The total number of data-points after removing the rows with missing values are:", len(customers))

The total number of data-points after removing the rows with missing values are: 1000


In [154]:
products.isnull().sum()

id          0
name        0
category    0
price       0
dtype: int64

##### From the above output, we can conclude and note that:
    - there is not missing values
    - There are some categorical features in our data frame; as there are some features in dtype: object). 
      So we will need toencode them into numeric forms later(customerID,country,gender,profession).

### Encoding categorical values 

In [155]:
numeric_features=customers.select_dtypes(include=['int64','float64']).columns
categorical_features=customers.select_dtypes(include='object').columns
from sklearn.preprocessing import LabelEncoder
scale=LabelEncoder()
for i in categorical_features :
    customers[i]=scale.fit_transform(customers[i])
customers.head()

id  country  gender  age  profession
0   0        2       0   42           2
1   1        5       0   41           1
2   2        5       0   42           1
3   3        1       0   39           2
4   4        0       0   43           3

# Cleaning Products dataset 

In [156]:
products.head()

id                                               name   category  \
0  PT-0001  Dual-Band / Tri-Mode Sprint PCS Phone w/ Voice...  High-Tech   
1  PT-0002                                Motorola I265 phone  High-Tech   
2  PT-0003  MOTOROLA C168i AT&T CINGULAR PREPAID GOPHONE C...  High-Tech   
3  PT-0004              Motorola i335 Cell Phone Boost Mobile  High-Tech   
4  PT-0005      Motorola V365 no contract cellular phone AT&T  High-Tech   

   price  
0    700  
1    650  
2    890  
3    900  
4    400

In [157]:
products.describe()

price
count  794.000000
mean   513.532746
std    280.395833
min     75.000000
25%    210.000000
50%    600.000000
75%    740.000000
max    900.000000

In [158]:
products.shape

(794, 4)

#### ====> The table of products have 794 rows and 4 columns.

### informations on features

In [159]:
products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 794 entries, 0 to 793
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id        794 non-null    object
 1   name      794 non-null    object
 2   category  794 non-null    object
 3   price     794 non-null    int64 
dtypes: int64(1), object(3)
memory usage: 24.9+ KB


### Missing values


In [160]:
products = products.dropna()
print("The total number of data-points after removing the rows with missing values are:", len(products))

The total number of data-points after removing the rows with missing values are: 794


In [161]:
products.isnull().sum()

id          0
name        0
category    0
price       0
dtype: int64

##### From the above output, we can conclude and note that:
    - there is not missing values
    - There are some categorical features in our data frame; as there are some features in dtype: object). 
      So we will need toencode them into numeric forms later(id,name,category).

### Encoding categorical values 


In [162]:
numeric_features1=products.select_dtypes(include=['int64','float64']).columns
categorical_features1=products.select_dtypes(include='object').columns
from sklearn.preprocessing import LabelEncoder
scale=LabelEncoder()
for i in categorical_features1 :
    products[i]=scale.fit_transform(products[i])
products.head()

id  name  category  price
0   0   116         0    700
1   1   287         0    650
2   2   247         0    890
3   3   321         0    900
4   4   313         0    400

# Cleaning Sales dataset

In [163]:
sales.head()

customerID productID  amount       Sdate
0    CT-0001   PT-0565   350.0  25/08/2020
1    CT-0001   PT-0565   350.0  12/10/2020
2    CT-0001   PT-0574   140.0  06/12/2021
3    CT-0001   PT-0574   140.0  19/02/2021
4    CT-0001   PT-0574   140.0  10/04/2021

In [164]:
sales.describe()

amount
count  3014.000000
mean    373.331453
std     286.479766
min      75.000000
25%     190.000000
50%     210.000000
75%     550.000000
max     990.000000

In [165]:
sales.shape

(3090, 4)

#### =====> The table of sales have 3090 rows and 4 columns.

### informations on features

In [170]:
sales.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3014 entries, 0 to 3014
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   customerID  3014 non-null   int32  
 1   productID   3014 non-null   int32  
 2   amount      3014 non-null   float64
 3   Sdate       3014 non-null   int32  
dtypes: float64(1), int32(3)
memory usage: 82.4 KB


### Missing values

In [171]:
sales.isnull().sum()

customerID    0
productID     0
amount        0
Sdate         0
dtype: int64

In [172]:
sales = sales.dropna()
print("The total number of data-points after removing the rows with missing values are:", len(sales))

The total number of data-points after removing the rows with missing values are: 3014


##### From the above output, we can conclude and note that:
    - there is missing values
    - There are some categorical features in our data frame; as there are some features in dtype: object).
    - we will need toencode them into numeric forms later(customerID,productID,Sdate).

### Encoding categorical values 

In [173]:
numeric_features2=sales.select_dtypes(include=['int64','float64']).columns
categorical_features2=sales.select_dtypes(include='object').columns
from sklearn.preprocessing import LabelEncoder
scale=LabelEncoder()
for i in categorical_features2 :
    sales[i]=scale.fit_transform(sales[i])
sales.head()

customerID  productID  amount  Sdate
0           1        219   350.0    670
1           1        219   350.0    315
2           1        228   140.0    159
3           1        228   140.0    492
4           1        228   140.0    247